lib/core/text/flat.nit

```diff
@@ -1184,10 +1183,11 @@ redef class NativeString
 		return new FlatString.with_infos(self, len, 0)
 	end
 
-	# Returns `self` as a new String.
-	redef fun to_s_with_copy: FlatString
+	redef fun to_s_with_copy do return to_s_with_copy_and_length(cstring_length)
+
+	# Get a `String` from `length` bytes at `self` copied into Nit memory
+	fun to_s_with_copy_and_length(length: Int): String
```

Comment:
With optional parameters in redefs we could even have a `to_s(length: nullable Int, copy: nullable Bool)`! But we would still need the long form method for foreign core where optional parameters are non-optional.